src/libtorch.cc

```diff
@@ -62,6 +62,107 @@
 
 namespace triton { namespace backend { namespace pytorch {
 
+namespace {
+
+// This object is used to capture timestamps for the instance group of type
+// 'MODEL'.
+class TimestampData {
+ public:
+  TimestampData()
+      : compute_input_start_(0), earliest_compute_infer_start_(0),
+        last_compute_infer_start_(0), compute_output_start_(0)
+  {
+  }
+
+  enum struct TimestampHint { COMPUTE_INPUT, COMPUTE_INFER, COMPUTE_OUTPUT };
+
+  // Set the timestamp hint
+  void SetTimestampHint(const TimestampHint& hint) { hint_ = hint; }
+
+  // Capture the timestamp based on the hint.
+  void CaptureTimestamp();
+
+  // Get the earliest timestamp.
+  void GetEarliestTimestamp(uint64_t& timestamp);
+
+  // Get the latest timestamp.
+  void GetLastTimestamp(uint64_t& timestamp);
+
+  // Get the duration of comput input.
+  uint64_t GetComputeInputDuration()
+  {
+    return last_compute_infer_start_ - compute_input_start_;
```

Comment:
For multi-device inference runs, wouldn't it be better to focus on capturing the complete compute_infer times? 
We should aim for `compute_input + compute_infer + compute_output` ==  `total_compute`?

I believe it would be better to capture: compute_input_start_, first_compute_infer_start_, last_compute_infer_end_.
The duration can be captured as:
compute_input = first_compute_infer_start_ - compute_input_start_
compute_infer = last_compute_infer_end_ - first_compute_infer_start_
compute_output = compute_output_end(last) - last_compute_infer_end_

I know this is not 100% accurate as a part of compute_input and compute_output might seep into compute_infer, but it is more comprehensible. 
@Tabrizian @krishung5 @nnshah1  what do you think?

Reply:
`first_compute_infer_start_` should not be that tricky as we synchronize all the streams before capturing this timestamp. Hence, most of these streams should capture identical timestamp. The `first_compute_infer_start_` will be the tricky one.

Reply:
With adding cuda stream synchronization before reading the outputs, we wouldn't need the `first_compute_infer_start_` and `last_compute_infer_end_`.